dipy/segment/tests/test_refine_rb.py

```diff
@@ -79,6 +81,33 @@ def test_rb_disable_slr():
         assert_equal(row.min(), 0)
 
 
+def test_rb_slr_threads():
+
+    rng_multi = np.random.RandomState(42)
+    rb_multi = RecoBundles(f, greater_than=0, clust_thr=10,
+                           rng=np.random.RandomState(42))
+    rec_trans_multi_threads, _ = rb_multi.recognize(model_bundle=f2,
+                                                    model_clust_thr=5.,
+                                                    reduction_thr=10,
+                                                    slr=True,
+                                                    slr_num_threads=None)
+
+    rb_single = RecoBundles(f, greater_than=0, clust_thr=10,
+                            rng=np.random.RandomState(42))
+    rec_trans_single_thread, _ = rb_single.recognize(model_bundle=f2,
+                                              model_clust_thr=5.,
+                                              reduction_thr=10,
+                                              slr=True,
+                                              slr_num_threads=1)
+
+    D = bundles_distances_mam(rec_trans_multi_threads, rec_trans_single_thread)
+
+    # check if the bundle is recognized correctly
+    # multi-threading prevent an exact match
+    for row in D:
```

Comment:
What type is D at this point? It's not an array? 

Do you understand why multi-threading prevents an exact match? Is there any additional randomness introduced within the algorithm, beyond the control you have via `rng`?

Reply:
Hey @frheault : I'd love to be able to merge this in. Any thoughts on these questions?

Reply:
D is an array, I used the same test "format" a every other recobundle test. I didn't want to diverge too much from the existing code. That's why I used this verification style.

The randomness is introduced from the optimizer ('L_BFGS_B' or 'Powell') from dipy.core.optimize import Optimizer which ultimately call scipy.optimize import fmin_l_bfgs_b, fmin_powell does not accept fixed initialization or seed.

rng only controls the shuffle and clustering of QBx, so the input to the optimizer is identical. The optimizer reach the same local minima, however the slight difference comes from numerical error due to different initialization for each thread.

Reply:
Gotcha. That all makes sense to me. The fact that scipy.optimize has randomization in it is very unfortunate, but out of our control here.

This all looks fine to me. I'm +1 for the merge here. 

Does anyone else want to take a look? If so, please do so in the next couple of days. I'll merge this mid next week, if no one complains before that.

Reply:
LGTM. I can merge if you want, or you can go ahead.

Thanks @frheault